pcap-rewrite/src/filters/filter_utils.rs

```diff
@@ -1,28 +1,34 @@
 use log::warn;
 
-use libpcap_tools::Error;
+use libpcap_tools::{Error, ParseContext};
 use pnet_packet::ethernet::{EtherTypes, EthernetPacket};
 use pnet_packet::vlan::VlanPacket;
 use pnet_packet::Packet;
 use pnet_packet::PrimitiveValues;
 
-pub fn extract_callback_ethernet<D>(
-    get_key_from_ipv4_l3_data: &dyn Fn(&[u8]) -> Result<D, Error>,
-    get_key_from_ipv6_l3_data: &dyn Fn(&[u8]) -> Result<D, Error>,
+// TODO: find simpler solution for function parameter with genericity
```

Comment:
While this works, static dispatch could be a problem because there are many key extraction functions, resulting in lots of code duplication et binary size increase. I think dynamic dispatch is useful here.
Here is an example of function signature for dynamic dispatch:
```rust
pub fn extract_callback_ethernet<D, F1, F2>(
    ctx: &ParseContext,
    get_key_from_ipv4_l3_data: &F1,
    get_key_from_ipv6_l3_data: &F2,
    packet_data: &[u8],
) -> Result<D, Error>
where
    F1: for<'a, 'b> Fn(&'a ParseContext, &'b [u8]) -> Result<D, Error>,
    F2: for<'a, 'b> Fn(&'a ParseContext, &'b [u8]) -> Result<D, Error>,
```